### PR TITLE
Add Storybook for UI iteration

### PR DESCRIPTION
**Pull request by r3mariano**
* Stories are .tsx files
* Storybook config reads from local misk js and css. This copies some work from @misk/dev, future work to avoid duplication?
* Added empty index files to make build pass

Other changes:
* Because of continued difficulties with PNPM on strict peer dependency
  mode, it was turned off
* PNPM and Rush versions were bumped
* Rush update was done to include new storybook dependencies

### CODE REVIEW

File: storybook/.storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const StyledComponentsTransformerPlugin = require("typescript-plugin-styled-components")
+const createStyledComponentsTransformer =
+  StyledComponentsTransformerPlugin.default
+module.exports = {
+  stories: ["../stories/**/*.tsx"],
+  addons: ["@storybook/addon-actions", "@storybook/addon-links"],
+  webpackFinal: (config) => {
+    const CopyWebpackPluginConfig = new CopyWebpackPlugin(
+      [
+        { from: "./node_modules/@misk/common/lib/web/" },
```
Comment (adrw) [inline storybook/.storybook/main.js:11]: Nice, this is the right way to do it 💯

File: storybook/.storybook/main.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        { from: "./node_modules/@misk/core/lib/web/" },
+        { from: "./node_modules/@misk/simpleredux/lib/web/" }
+      ],
+      { copyUnmodified: true }
+    )
+    const StyledComponentsTransformer = createStyledComponentsTransformer()
```
Comment (adrw) [inline storybook/.storybook/main.js:17]: We probably don't need this since Misk-Web has migrated to Emotion?

File: storybook/stories/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import React from "react"
+import { BrowserRouter } from "react-router-dom"
+import { action } from "@storybook/addon-actions"
+import { Navbar, Environment } from "@misk/core"
+
+export const Production = () => (
+  <BrowserRouter>
+    <Navbar
```
Comment (adrw) [inline storybook/stories/Navbar.tsx:8]: Really powerful to see how simple it is to build a new story to test out a UI component. Pretty excited at the thought of adding the rest of Misk-Web components here and potentially transitioning it to be the primary docs site.

File: storybook/stories/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        "Second Item"
+      ]}
+      status={"Status Text"}
+    />
+  </BrowserRouter>
+)
+
+export const Staging = () => (
+  <BrowserRouter>
+    <Navbar
```
Comment (adrw) [inline storybook/stories/Navbar.tsx:37]: Have you tested with this setup that changes you make to the Navbar in `packages/@misk/core/` appear in the Storybook once rebuilt? That would be the key part to allow use of the Storybook when hacking on local library changes.

File: storybook/.storybook/preview.js [hunk 5]
```diff
@@ -0,0 +1 @@
+import "@misk/common/lib/web/@misk/common/styles.css"
```
Comment (adrw) [inline storybook/.storybook/preview.js:1]: Cool 👍

**Comment by adrw** [comment]
Should the stories be `.tsx` files? Generally files that have inline jsx would end with the x. Using tsx generally plays nicer with IDE syntax highlighting so would be preferred if possible.

**Comment by r3mariano** [comment]
> Should the stories be `.tsx` files? Generally files that have inline jsx would end with the x. Using tsx generally plays nicer with IDE syntax highlighting so would be preferred if possible.

Yes they are. I'll update the description to match.

File: storybook/stories/Navbar.tsx (hunk 4 above)
Comment (r3mariano) [reply storybook/stories/Navbar.tsx:37]: Good point, will try it out.

File: storybook/.storybook/main.js (hunk 2 above)
Comment (r3mariano) [reply storybook/.storybook/main.js]: I've removed it

File: storybook/stories/Navbar.tsx (hunk 4 above)
Comment (r3mariano) [reply storybook/stories/Navbar.tsx:37]: It definitely works ✨  

It watches the lib folders and hot-reloads the components.

Comment (adrw) [reply storybook/stories/Navbar.tsx:37]: So cool! Let’s merge!

Comment (r3mariano) [reply storybook/stories/Navbar.tsx:37]: I have no merge access :(

I'm also not very familiar with whether we should squash this PR into one commit or merge as is.

Comment (adrw) [reply storybook/stories/Navbar.tsx:37]: Github squash feature works great! I'll merge 😊

**Pull request merged**